### PR TITLE
feat: integration feature flags

### LINKED ISSUE #14: Integrate with our Feature Flags API
## New contributors welcome!

Hey 👋  are you new around here? Feel free to contribute to this issue! It is quite straightforward since there's a clear spec - no need to think about what's the best solution. 

However, it isn't a teeny tiny task, which is why we're offering a **$100 gift card** for [PostHog merch](https://merch.posthog.com/) for a merged PR that implements this. You'll also be **listed as a contributor** on [our main repo's README (4.3k stars)](https://github.com/PostHog/posthog#contributors-) and get a [digital contributor card](https://posthog.com/contributors).

## Description

PostHog provides support for [feature flags](https://posthog.com/docs/features/feature-flags). These allow turning functionality on and off based on a response from the PostHog API.

For a client-side library like this one, we'd need an implementation similar to that of [`posthog-js`](https://posthog.com/docs/libraries/js#feature-flags).

Here's an overview of what implementing this entails:

- A mechanism to send a `POST` request to `/decide/` with the following data:

    ```js
    // Content-Type: application/json
    {
        token: <posthog_project_api_key>,
        distinct_id: <user_distinct_id>
    }
    ```
    The response will contain an array/list at `response['featureFlags']` with the flags that are enabled for this user. The distinct ID should be already available from existing methods implemented in this lib.
    The response should be cached and refreshed periodically.

- A method `isFeatureEnabled`

    This method takes a feature flag key as an input and returns a boolean indicating if the given flag is on or off for the user based on the response from above.
    
- A method `reloadFeatureFlags`

    This forces a refresh of the cache by calling `/decide/` again.

- A method `onFeatureFlagsLoaded`

    Every time the flags are reloaded, this should be called as a callback.

## Questions?

Feel free to ask me anything about this! I'm also reachable on the [PostHog Slack](https://posthog.com/slack).

### PR DESCRIPTION
**Pull request by Kiruel**
This PR will fix #14 

**This PR need to have:**

- [x] A new release for https://github.com/PostHog/posthog-ios , ~~currently only available in master.~~
- [x] Same for android https://github.com/PostHog/posthog-android.

**WIP:**
- [x] Android integration
    - [x] `isFeatureEnabled`
    - [x] `reloadFeatureFlags`
- [x] iOS integration
    - [x] `isFeatureEnabled`
    - [x] `reloadFeatureFlags`
- [x] Web integration
    - [x] `isFeatureEnabled`
    - [x] `reloadFeatureFlags`
    
[Slack thread](https://app.slack.com/client/TT68H7WHY/C019E06QDG8/thread/C019E06QDG8-1673344568.244509)

### CODE REVIEW

**Comment by aylton-almeida** [comment]
Is there any update on this? I'd like to help somehow if possible.

**Comment by Kiruel** [comment]
Hi @aylton-almeida , I would like to continue this task but I need some time. Probably next week I will keep working on it.

File: android/build.gradle
Comment (Kiruel) [inline android/build.gradle:38]: Here waiting for release android SDK 2.0.0

Comment (Kiruel) [reply android/build.gradle]: @joethreepwood resolved

**Comment by Kiruel** [comment]
@joethreepwood We should be good for a review now, android version is 2.0.0.

**Comment by neilkakkar** [comment]
Hey @Kiruel , thanks for the contribution! Sorry this is taking ages to review (it's not usual at PostHog). It's happening here because we don't have flutter knowledge and need someone to get up to speed so we can review 😅

**Comment by neilkakkar** [comment]
But, the good news is that since all dependencies here are now complete, we're prioritising this, so you should definitely have a review by next week!

**Comment by Kiruel** [comment]
@neilkakkar Hi, anynews ?

File: android/.settings/org.eclipse.buildship.core.prefs
Comment (liyiy) [inline android/.settings/org.eclipse.buildship.core.prefs:4]: curious why we are updating the versions here and below?

**Review by liyiy — APPROVED** [review]
I am no Flutter expert but this seems like a basic enough implementation of feature flags

File: android/.settings/org.eclipse.buildship.core.prefs
Comment (liyiy) [reply android/.settings/org.eclipse.buildship.core.prefs:4]: @Kiruel

Comment (Kiruel) [reply android/.settings/org.eclipse.buildship.core.prefs:4]: Probably my vscode doing weird things, should I remove this ?

Comment (liyiy) [reply android/.settings/org.eclipse.buildship.core.prefs:4]: If it's not needed then sure! 😃

Comment (Kiruel) [reply android/.settings/org.eclipse.buildship.core.prefs]: @liyiy Pushed without it.

**Comment by liyiy** [comment]
awesome! @Kiruel would you like to update the changelog and release for this too?

**Comment by Kiruel** [comment]
Thanks @liyiy for your time on this one ! I already change the changelog for the documentation, and I think I don't have the right to create a release.

**Pull request merged**
